packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -2395,121 +2060,56 @@ class CupertinoDialogAction extends StatelessWidget {
 
 // iOS style dialog action button layout.
 //
-// [_CupertinoDialogActionsRenderWidget] does not provide any scrolling
+// [_AlertDialogActionsLayout] does not provide any scrolling
 // behavior for its buttons. It only handles the sizing and layout of buttons.
 // Scrolling behavior can be composed on top of this widget, if desired.
 //
-// See [_RenderCupertinoDialogActions] for specific layout policy details.
-class _CupertinoDialogActionsRenderWidget extends MultiChildRenderObjectWidget {
-  const _CupertinoDialogActionsRenderWidget({
-    required List<Widget> actionButtons,
-    double dividerThickness = 0.0,
-    bool hasCancelButton = false,
-  }) : _dividerThickness = dividerThickness,
-       _hasCancelButton = hasCancelButton,
-       super(children: actionButtons);
+// The layout operates in two modes:
+//
+// 1. Horizontal Mode: If there are exactly two buttons and they fit in a single
```

Comment:
> I see the logic is kinda inherited from _RenderCupertinoDialogActions

Nah, the behavior is observed on iOS simulator, and the logic is what I consider the simplest way to describe the behavior.

> build the widget tree based on the number of buttons (either Column or OverflowBar)

I would if the two-button layout logic can be described as purely widgets (including `OverflowBar`). However, since now I have yet to find such a way, and so that I had to write a custom layout widget, I want to at least reuse as much existing code as possible. Now that the vertical layout is simply a `Column` for two buttons, `_AlertDialogActionsLayout` will inherit `Column` anyway, why not make `_AlertDialogActionsLayout` handle the other cases, which are also just a `Column`?